HdProduction.HelpDesk.Infrastructure/Services/UserService.cs

```diff
@@ -14,12 +15,20 @@ public UserService(IUserRepository userRepository)
             _userRepository = userRepository;
         }
 
-        public Task CreateAsync(string email, string pwdHash)
+        public async Task<User> CreateAsync(string firstName, string lastName, string email, string pwdHash)
         {
             var pwdHelper = SecurityHelper.Create();
-            _userRepository.Add(new User(email, "abc", "cde", "", 
-                pwdHelper.CreateSaltedPassword(pwdHash), pwdHelper.Salt));
-            return _userRepository.SaveAsync();
+            var user = new User(email, firstName, lastName, "",
+                pwdHelper.CreateSaltedPassword(pwdHash), pwdHelper.Salt);
+            _userRepository.Add(user);
```

Comment:
add class UserSafeguard : IUserSafeguard
with validations like

void EnsureEmail(string email){
  if(regex.NotMatch(email)){
    throw new BusinessLogic("Not valid email");
  }
  if (_userRepository.FindByEmail(email) != null){
     throw new BusinessLogic("Already exist");
  }
}

Reply:
I would like to see it in your example:)

Reply:
added commit with example

Reply:
Let keep it as is for now. But actually, I don't like this approach. But let discuss it later.